libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -20,17 +20,15 @@ const AP_Param::GroupInfo AP_DDS_Client::var_info[]= {
 
 #include "AP_DDS_Topic_Table.h"
 
-void AP_DDS_Client::update_topic(builtin_interfaces_msg_Time* msg)
+void AP_DDS_Client::update_topic(builtin_interfaces_msg_Time& msg)
 {
-    if (msg != nullptr) {
-
-        uint64_t utc_usec;
-        if (!AP::rtc().get_utc_usec(utc_usec)) {
-            utc_usec = AP_HAL::micros64();
-        }
-        msg->sec = utc_usec / 1000000ULL;
-        msg->nanosec = (utc_usec % 1000000ULL) * 1000UL;
+    uint64_t utc_usec;
+    if (!AP::rtc().get_utc_usec(utc_usec)) {
+        utc_usec = AP_HAL::micros64();
     }
+    msg.sec = utc_usec / 1000000ULL;
+    msg.nanosec = (utc_usec % 1000000ULL) * 1000UL;
```

Comment:
```suggestion
    msg = {
        utc_usec / 1000000ULL,
        (utc_usec % 1000000ULL) * 1000UL
    };
```

Just a suggestion that you play with this.  It means the structure is fully initialised on return, and gives you some type-checking too.

Reply:
It sadly loses the context of the first element being seconds and the second element being millis. Unless you could use C++20's [aggregate initialization that supports designated initializers]( https://en.cppreference.com/w/cpp/language/aggregate_initialization). Maybe there a way to get the best of both worlds 🔍